formatTest/unit_tests/expected_output/basicStructures.re

```diff
@@ -691,6 +691,11 @@ type component2 = {
   updater: unit
 };
 
+type component3 = {
+  props: M.props,
```

Comment:
Should the types also be punned?

Reply:
cc @hcarty. What do you think? I didn't do it because `{M.x: x, y: 1}` has the `Foo` part on the left and it was less intuitive punning the type when `M` is on the right

Reply:
We're referencing the `M.t` type here for use as a field here, not extending the type itself. I agree with @chenglou that it would be better to not allow punning in this case. 

That may be worth considering as syntax for the 4.06.0 feature supporting inherited fields in object types.

Reply:
Oh I see I see. Good point.